cesmcatalog/gen_CESM_catalog.py

```diff
@@ -0,0 +1,224 @@
+    logger = logging.getLogger(__name__)
+    logger.info('Will catalog files in {}'.format(archive_root))
+
+
+################################################################################
+
+
+def _gen_timeseries_catalog(case_root, archive_root, run_config):
+    import pandas as pd
+    import xarray as xr
+
+    # Set up logger
+    # Define casename, file to create, and columns the catalog will contain
+    logger = logging.getLogger(__name__)
+    casename = run_config['CASE']
+    out_file = 'cesm_catalog.csv.gz'
+    col_order = [
+        'case',
+        'component',
+        'stream',
+        'variable',
+        'long_name',
+        'start_date',
+        'end_date',
+        'path',
+        'parent_branch_year',
+        'child_branch_year',
+        'parent_case',
+    ]
+    long_name_dict = dict()
+
+    # cd archive_root and make sure intake/ subdirectory exists (for output)
+    os.chdir(archive_root)
+    if not os.path.isdir('intake'):
+        os.mkdir('intake')
+
+    # want paths in catalog to be relative to location of catalog
+    os.chdir('intake')
+    logger.info('Will catalog files in {}'.format(archive_root))
+    catalog = dict()
+    for col_name in col_order:
+        catalog[col_name] = []
+
+    # Find each netcdf file in directory
+    for root, dirs, files in os.walk('..'):
+        for ncfile in fnmatch.filter(files, '{}*.nc'.format(casename)):
+            # each file should be {casename}.{stream}.{variable}.{start_date}-{end_date}.nc
+            # first we drop the leading {casename}.
+            file_without_case = ncfile.replace(casename, '')[1:]
+            # then we split on .
+            file_info = file_without_case.split('.')
+            # {stream} will have at least one . in it
+            # figure out how many by location of date_range
+            # (only part of filename that should contain a '-')
+            for date_ind, info in enumerate(file_info):
+                if len(info.split('-')) > 1:
+                    break
+
+            # Enough to determine stream, variable, start_date, and end_date
+            catalog['stream'].append('.'.join(file_info[: date_ind - 1]))
+            variable = file_info[date_ind - 1]
+            catalog['variable'].append(variable)
+            date_range = info.split('-')
+            catalog['start_date'].append(date_range[0])
+            catalog['end_date'].append(date_range[1])
+
+            # path should be relative to intake/, so we keep root
+            path = os.path.join(root, ncfile)
+            catalog['path'].append(path)
+            # component is the name of the subdirectory of archive_root
+            component = path.split('/')[1]
+            catalog['component'].append(component)
+            if component not in long_name_dict:
+                long_name_dict[component] = dict()
+            if variable not in long_name_dict[component]:
+                ds = xr.open_dataset(path, decode_cf=False)
```

Comment:
Using dask by specifying `chunks={}` may improve performance:

```python
In [1]: import xarray as xr

In [2]: %time xr.tutorial.open_dataset('rasm', decode_times=False)
CPU times: user 6.08 ms, sys: 9.71 ms, total: 15.8 ms
Wall time: 452 ms

In [3]: %time xr.tutorial.open_dataset('rasm', decode_times=False, chunks={})
CPU times: user 3.29 ms, sys: 873 µs, total: 4.17 ms
Wall time: 17.8 ms
```

```suggestion
                ds = xr.open_dataset(path, decode_cf=False, chunks={})
```

Reply:
Done via e091854d, thanks! I'm not seeing a performance boost yet, but I wonder if it'll speed things up once we get to bigger time series files.